### PR TITLE
[@types/mongodb] make _id optional for collection.insertX functions

### LINKED ISSUE #33740: Letting mongodb pick the _id is not possible anymore when typing collections.
- [x] I tried using the `@types/mongodb` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - @CaselIT
  - @alanmarcell
  - @bitjson
  - @dante-101
  - @mcortesi
  - @EnricoPicci
  - @AJCStriker
  - @julien-c
  - @daprahamian
  - @denys-bushulyak
  - @BastienAr
  - @sindbach
  - @geraldinelemeur
  - @jishi
  - @various89
  - @angela-1
  - @lirbank
  - @hector7
  - @floric
  - @erikc5000
  - @Manc

Since [this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29245)
`Collection.insert{,One,Many}` are typed with TSchema from `Collection<TSchema>`
same as the return type of the query methods. This makes it a bit hard to use
when you want mongodb to pick the _id of an inserted document for you:

Version with `_id` being part of the TSchema:
```js
import { Collection, MongoClient, ObjectId } from 'mongodb'

interface Person {
  _id: ObjectId
  name: string 
  age: number
}

MongoClient.connect().then(async client => {
  const db = client.db()
  const personCollection: Collection<Person> = db.collection('persons');
  
  const writeResult = await personCollection.insertOne({ name: "Foo Bar", age: 30 }) // type error, `_id` missing
  
  const person = await personCollection.findOne({ _id: writeResult.insertedId })
  
  if (person) {
    console.log(person._id);
  }
})
```

Version without `_id` being part of the TSchema:
```js
import { Collection, MongoClient, ObjectId } from 'mongodb'

interface Person {
  name: string 
  age: number
}

MongoClient.connect().then(async client => {
  const db = client.db()
  const personCollection: Collection<Person> = db.collection('persons');
  
  const writeResult = await personCollection.insertOne({ name: "Foo Bar", age: 30 }) 
  
  const person = await personCollection.findOne({ _id: writeResult.insertedId })
  
  if (person) {
    console.log(person._id); // type error, _id is not defined
  }
})
```

In either case it's not possible to sensibly type collections anymore :(

I think it would make sense to make the `_id` property optional for the inserts.
I can make a PR I just wanted to ask for feedback first.

### PR DESCRIPTION
**Pull request by HosseinAgha**
bumps typescript version to 2.8

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://mongodb.github.io/node-mongodb-native/3.2/api/Collection.html#insert>>

Trying to continue the good work done in #37196

This PR marks the _id property as optional any for all Collection<T>.insert statements.

---

resolves #33740

### CODE REVIEW

**Comment by HosseinAgha** [comment]
Hi @CaselIT.  
As you suggested in #37196 I used a `Pick` to create `Omit` and bumped the required typescript version to 2.8 but my tests still fail with. I do not understand the error __"agenda depends on mongodb but has a lower required TypeScript version"__.  
Could you please guide me?

**Comment by CaselIT** [comment]
Hi @HosseinAgha 

> agenda depends on mongodb but has a lower required TypeScript version

I think that you need to also bump the version in agenda here 
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/agenda

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -36,6 +37,8 @@ import { EventEmitter } from 'events';
 import { Readable, Writable } from "stream";
 import { checkServerIdentity } from "tls";
 
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
```
Comment (CaselIT) [inline types/mongodb/index.d.ts]: We can maybe add a comment to indicate that this is needed until the typescript version is lower than 3.5

**Review by CaselIT — APPROVED** [review]
Looks ok.
I've just added a suggestion

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (HosseinAgha) [reply types/mongodb/index.d.ts]: Done

**Pull request merged**